bin/metrics-cgroup.rb

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env ruby
+#
+#   metrics-cgroup
+#
+# DESCRIPTION:
+#   This plugin collects cgroup metrics.
+#
+# OUTPUT:
+#   metrics data in graphite
+#
+# PLATFORMS:
+#   Linux
+#
+# DEPENDENCIES:
+#   gem: sensu-plugin
+#
+# USAGE:
+#   # Collect all cgroup metrics, tested only on CentOS.
+#   metrics-cgroup.rb -g "/sys/fs/cgroup/**/**"
+#
+#   # Collect only systemd service metrics
+#   metrics-cgroup.rb -g "/sys/fs/cgroup/**/system.slice/*"
+#
+#   # Collect only user.slice only
+#   metrics-cgroup.rb -g "/sys/fs/cgroup/**/user.slice"
+#
+#   # Collect only mesos task metrics
+#   metrics-cgroup.rb -g "/sys/fs/cgroup/**/mesos/*"
+#
+#   # Collect only docker container metrics
+#   # (may not be meaningful if you already collect metrics from docker)
+#   metrics-cgroup.rb -g "/sys/fs/cgroup/**/docker/*"
+#
+#   # collect only system.slice, user.slice, docker and mesos and nothing else
+#   metrics-cgroup.rb -g "/sys/fs/cgroup/**/{system.slice,user.slice,docker,mesos}{,/*}"
+#
+#   # Instead of collecting breakdown of each docker instance, or
+#   # mesos task, or systemd service metric; this could let to
+#   # collect/compare overall docker, mesos, systemd and user recources
+#   # in total.
+#   metrics-cgroup.rb -g "/sys/fs/cgroup/**/{system.slice,user.slice,docker,mesos}{,/*}"
+#
+# LICENCE:
+#   Released under the same terms as Sensu (the MIT license); see LICENSE
+#   for details.
+#
+require 'sensu-plugin/metric/cli'
+require 'socket'
+
+class CgroupMetrics < Sensu::Plugin::Metric::CLI::Graphite
+  FILES_TO_SKIP = [
```

Comment:
Lets expose this as an option, I have seen many times where a new version gets released and people are forced to wait until it can be fixed and released when ideally they should be able to overwrite this behavior. Here is an example from the redis one: https://github.com/sensu-plugins/sensu-plugins-redis/blob/2.0.0/bin/metrics-redis-graphite.rb#L121-L125